src/txn/txn.c

```diff
@@ -1331,8 +1333,10 @@ __wt_txn_rollback(WT_SESSION_IMPL *session, const char *cfg[])
 
 		__wt_txn_op_free(session, op);
 	}
-	WT_ASSERT(session, skip_update_assert ||
-	    resolved_update_count == visited_update_count);
+	WT_RET_ASSERT(session, skip_update_assert ||
+	    resolved_update_count == visited_update_count, EINVAL,
+	    "Expected number of resolved prepared updates does not"
+	    " match actual");
```

Comment:
If we have a varargs here, why are we not giving the user the expected and actual values that failed the assert?